### PR TITLE
Add eip

### PR DESCRIPTION
**Pull request by Joel-Cripps**
This PR adds the ability to have a single static address for the Primary instance. It works by  first deleting the ephemeral external address created by GCP and then injecting a new into the access config. 

There is not exactly an EIP in GCP but this will work in a similar way by moving a static address with the new primary instances.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -354,6 +361,9 @@ resource "google_compute_forwarding_rule" "internal_load_balancer" {
   network               = "${google_compute_network.protected_vpc_network.self_link}"
   subnetwork            = "${google_compute_subnetwork.protected_subnet.self_link}"
 }
+resource "google_compute_address" "master_external_address" {
+  name = "${var.cluster_name}-external-address-${random_string.random_name_post.result}"
+}
```
Comment (JaydenLiang) [inline main.tf:367]: I am just curious if there's resource name length limit in your resources?

**Review by JaydenLiang — APPROVED** [review]
looks good to me. I left a comment regarding resource name length limit.

File: main.tf [hunk 2]
```diff
@@ -392,3 +397,9 @@ output "Trigger_URL" {
 output "Note" {
   value = "The FireStore Database must be deleted separately"
 }
+output "Master_Static_Address" {
```
Comment (jamie-pate) [inline main.tf]: primary?

**Review by jamie-pate — CHANGES_REQUESTED** [review]

File: main.tf (hunk 2 above)
Comment (Joel-Cripps) [reply main.tf]: Thanks, fixed this in the next commit.

File: main.tf (hunk 1 above)
Comment (Joel-Cripps) [reply main.tf:367]: If you attempt to put one that is too large Terraform will generate an error like so :

```
Error: "name" ("fortigateautoscale-externaaslkdalsdkjflasdjflasjdflaklasdhjfalskdhfjlaksdhfkalsdhfklasdhfaksldhfaksldhfksdjhfskdjhfaksjdhfkashdfkhfaskldhfasdfl-address-toxuy") doesn't match regexp "^(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?)$"
```

**Review by jamie-pate — APPROVED** [review]

**Comment by Joel-Cripps** [comment]
Fixed the issue with the delete sometimes returning a "RUNNING" status. Will now check for up to a max of 45 seconds to see if delete was finished.

File: index.ts [hunk 3]
```diff
@@ -401,7 +403,7 @@ export class GCP extends CloudPlatform<
         // First must delete the ephemeral address on a host.
         // Then we can add the EIP(or static address in the case of GCP)
         try {
-            const deleteReq = await client.request({
+            const deleteReq:any = await client.request({
```
Comment (JaydenLiang) [inline index.ts]: looks like deleteReq is an object  ` {data:{status:string, [key:string]:string},[key:string]:string}`. Can use it to replace type 'any'.

File: index.ts [hunk 4]
```diff
@@ -410,11 +412,33 @@ export class GCP extends CloudPlatform<
...
             });
             // Troubleshooting data:
-            console.log(`Delete req ${JSON.stringify(deleteReq.data)}`);
+            console.log(`Delete EIP request ${JSON.stringify(deleteReq.data)}`);
+            // If the request has been queued check the VM for the current accessConfig
+            if(deleteReq.data.status !== 'SUCCEEDED'){
+                console.warn(`Delete has queued. Waiting for it to finish before adding Static IP`)
+                let running = true;
+                let startTime = Date.now();
+                const attachEIPTimeLimit = parseInt(ATTACH_EIP_TIME_LIMIT) || 45000 // Default 45 seconds max.
```
Comment (JaydenLiang) [inline index.ts]: can use a script level const DEFAULT_ATTACH_EIP_TIME_LIMIT = 4500 for this magic number.

**Review by JaydenLiang — CHANGES_REQUESTED** [review]

File: index.ts (hunk 4 above)
Comment (JaydenLiang) [reply index.ts]: my comment refers to the 4500 in line 421.

File: index.ts (hunk 3 above)
Comment (Joel-Cripps) [reply index.ts]: it is an object, but a pretty complicated one for the entire VM data, that changes a lot based on what vm etc. I could replace the return with the info I need instead.

Comment (Joel-Cripps) [reply index.ts]: Oh sorry nevermind.  I misread. Yeah I should be able to make a type for deleteReq

File: index.ts (hunk 4 above)
Comment (Joel-Cripps) [reply index.ts]: Sure, Either way works for me.

File: index.ts (hunk 3 above)
Comment (Joel-Cripps) [reply index.ts]: Added type for deleteReq
```    
export interface DeleteRequest {
        data: Data;
      }
      export interface Data {
        id: string;
        name: string;
        zone: string;
        operationType?: string;
        targetLink?: string;
        targetId?: string;
        status?: string;
        user?: string;
        progress?: number;
        insertTime?: string;
        startTime?: string;
        selfLink: string;
        kind?: string;
      }
      ```

File: index.ts (hunk 4 above)
Comment (Joel-Cripps) [reply index.ts]: No longer process.env now just:
ATTACH_EIP_TIME_LIMIT = 45000

**Review by JaydenLiang — APPROVED** [review]

**Review by jamie-pate — APPROVED** [review]

**Pull request merged**